odc/geo/_xr_interop.py

```diff
@@ -123,21 +123,16 @@ def spatial_dims(
 def _mk_crs_coord(crs: CRS, name: str = "spatial_ref") -> xarray.DataArray:
     # pylint: disable=protected-access
 
-    if crs.projected:
-        grid_mapping_name = crs._crs.to_cf().get("grid_mapping_name")
-        if grid_mapping_name is None:
-            grid_mapping_name = "??"
-        grid_mapping_name = grid_mapping_name.lower()
-    else:
-        grid_mapping_name = "latitude_longitude"
-
+    cf = crs.proj.to_cf()
+    cf.setdefault("grid_mapping_name", "??")
```

Comment:
Is there a reason `??` is used? Should the be a constant somewhere? Or is this just for printing purposes?

Reply:
Actually I'm not sure we should be setting it to anything. Basically `grid_mapping_name` is an `enum` defined by CF convention, so if your CRS is not supported by CF convention, it should probably not have `grid_mapping_name` defined at all. We are relying on it being there do detect "CRS coordinates" attached to an xarray. But this should probably change to look for any of these attributes instead: `grid_mapping_name, spatial_ref, crs_wkt`.

Not sure what that means for compatibility of files produced with `.to_netcdf(...)`. 


https://cfconventions.org/Data/cf-conventions/cf-conventions-1.7/build/apf.html

Reply:
hm, looks like we are not even using that attribute for locating crs coordinate. I think it should just be left unset if it's not defined in the output of `.to_cf()`.

I should also add tests for round-tripping CRS to netcdf/zarr/tiff. it could be that non-standard name is worse than no name.